src/GraphQL/Schemas/Primary/Substrate/Mutations/InfuseMutation.php

```diff
@@ -0,0 +1,142 @@
+use Illuminate\Support\Arr;
+use Rebing\GraphQL\Support\Facades\GraphQL;
+
+class InfuseMutation extends Mutation implements PlatformBlockchainTransaction, PlatformGraphQlMutation
+{
+    use HasEncodableTokenId;
+    use HasIdempotencyField;
+    use HasSigningAccountField;
+    use HasSimulateField;
+    use HasSkippableRules;
+    use HasTokenIdFieldRules;
+    use HasTokenIdFields;
+    use HasTransactionDeposit;
+    use InPrimarySubstrateSchema;
+    use StoresTransactions;
+
+    /**
+     * Get the mutation's attributes.
+     */
+    public function attributes(): array
+    {
+        return [
+            'name' => 'Infuse',
+            'description' => __('enjin-platform::mutation.infuse.description'),
+        ];
+    }
+
+    /**
+     * Get the mutation's return type.
+     */
+    public function type(): Type
+    {
+        return GraphQL::type('Transaction!');
+    }
+
+    /**
+     * Get the mutation's arguments definition.
+     */
+    public function args(): array
+    {
+        return [
+            'collectionId' => [
+                'type' => GraphQL::type('BigInt!'),
+                'description' => __('enjin-platform::mutation.common.args.collectionId'),
+            ],
+            ...$this->getTokenFields(__('enjin-platform::mutation.infuse.args.tokenId')),
+            'amount' => [
+                'type' => GraphQL::type('BigInt!'),
+                'description' => __('enjin-platform::mutation.infuse.args.amount'),
+            ],
+            ...$this->getSigningAccountField(),
+            ...$this->getIdempotencyField(),
+            ...$this->getSkipValidationField(),
+            ...$this->getSimulateField(),
+        ];
+    }
+
+    /**
+     * Resolve the mutation's request.
+     */
+    public function resolve(
+        $root,
+        array $args,
+        $context,
+        ResolveInfo $resolveInfo,
+        Closure $getSelectFields,
+        SerializationServiceInterface $serializationService,
+        TransactionService $transactionService,
+    ): mixed {
+        $encodedData = $serializationService->encode($this->getMutationName(), static::getEncodableParams(
+            collectionId: $args['collectionId'],
+            tokenId: $this->encodeTokenId($args),
+            amount: $args['amount']
+        ));
+
+        return Transaction::lazyLoadSelectFields(
+            $this->storeTransaction($args, $encodedData),
+            $resolveInfo
+        );
+    }
+
+    public static function getEncodableParams(...$params): array
+    {
+        return [
+            'collectionId' => gmp_init(Arr::get($params, 'collectionId', 0)),
+            'tokenId' => gmp_init(Arr::get($params, 'tokenId', 0)),
+            'amount' => gmp_init(Arr::get($params, 'amount', 0)),
+        ];
+    }
+
+    /**
+     * Get the mutation's validation rules.
+     */
+    protected function rulesWithValidation(array $args): array
+    {
+        return [
+            // TODO: Check anyoneCanInfuse
```

Comment:
what's the final conclusion for this?

Reply:
No final conclusion, it will be done later after the release